test/parallel/test-process-env-allowed-flags-are-documented.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+const common = require('../common');
+
+if (process.config.variables.node_without_node_options) {
+  common.skip('missing NODE_OPTIONS support');
+}
+
+const assert = require('assert');
+const fs = require('fs');
+const path = require('path');
+
+const docsDir = path.resolve(__dirname, '..', '..', 'doc');
+const cliMd = path.join(docsDir, 'api', 'cli.md');
+const cliDocs = fs.readFileSync(cliMd, { encoding: 'utf8' });
+const docRegExp = /^Node\.js options that are allowed are:$([^#]+)/m;
```

Comment:
This seems brittle. A line-wrap change could break it. And I'll definitely break it when I inevitably decide to reduce passive voice from the doc. But I don't have a better idea, so...¯\\_(ツ)_/¯  (FWIW, I'd prefer if the doc said something like "Valid options are:". But obviously that can and likely should be some other PR some time.)

Reply:
FWIW if we change the line in `doc/api/cli.md` to `Valid options are:` the test fails:

```console
-bash-4.2$ ./node test/parallel/test-process-env-allowed-flags-are-documented.js
/home/users/riclau/sandbox/github/nodejs/test/parallel/test-process-env-allowed-flags-are-documented.js:17
const docOptions = cliDocs.match(docRegExp)[0];
                                           ^

TypeError: Cannot read property '0' of null
    at Object.<anonymous> (/home/users/riclau/sandbox/github/nodejs/test/parallel/test-process-env-allowed-flags-are-documented.js:17:44)
    at Module._compile (internal/modules/cjs/loader.js:779:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:790:10)
    at Module.load (internal/modules/cjs/loader.js:642:32)
    at Function.Module._load (internal/modules/cjs/loader.js:555:12)
    at Function.Module.runMain (internal/modules/cjs/loader.js:842:10)
    at internal/main/run_main_module.js:17:11
-bash-4.2$
```

I'll see what we can do to make this better.

Reply:
@Trott Changed this so that it now starts at the section header, which is less likely to change. PTAL.

(Test now still works when the prose is changed to "Valid options are:" but I'll let you make that change.)

Reply:
Since HTML comments are valid Markdown, maybe we can delineate the list with a self-documenting HTML comment and use a RegExp to search for that? A suggestion, but totally not blocking this.

Reply:
Added HTML comment markers. PTAL.

Of course now we have markers the follow on step would be to automatically update the list (since Node.js know which options are allowed in the environment and we've marked in the document where the lists should go). I'd prefer that to be a follow on PR 😁.